pulsar-client-cpp/include/pulsar/Auth.h

```diff
@@ -26,9 +26,27 @@
 #pragma GCC visibility push(default)
 
 namespace pulsar {
-    
-    typedef std::map<std::string, std::string> ParamMap;
-    
+
+    // for compatibility with previous version
+    enum AuthType {
```

Comment:
This good too. In addition, can you also update the comment to say this is deprecated?